smithy-model/src/main/java/software/amazon/smithy/model/validation/validators/HttpMethodSemanticsValidator.java

```diff
@@ -45,20 +45,25 @@
      *     <li>(Boolean) isReadonly</li>
      *     <li>(Boolean) isIdempotent</li>
      *     <li>(Boolean) allowsRequestPayload</li>
+     *     <li>(Boolean) warningMessageWhenModeled</li>
      * </ul>
      *
      * <p>Setting any of the Boolean properties to {@code null} means that
      * the validator will have no enforced semantic on a specific property.
      */
     private static final Map<String, HttpMethodSemantics> EXPECTED = MapUtils.of(
-            "GET", new HttpMethodSemantics(true, false, false),
-            "HEAD", new HttpMethodSemantics(true, false, false),
-            "OPTIONS", new HttpMethodSemantics(true, false, false),
-            "TRACE", new HttpMethodSemantics(true, false, false),
-            "POST", new HttpMethodSemantics(false, null, true),
-            "DELETE", new HttpMethodSemantics(false, true, false),
-            "PUT", new HttpMethodSemantics(false, true, true),
-            "PATCH", new HttpMethodSemantics(false, null, true));
+            "GET", new HttpMethodSemantics(true, false, false, null),
+            "HEAD", new HttpMethodSemantics(true, false, false, null),
+            "OPTIONS", new HttpMethodSemantics(true, false, true,
+                    "OPTIONS requests are typically used as part of CORS and should not be modeled explicitly. They "
+                    + "are an implementation detail that should not appear in generated client or server code. "
+                    + "Instead, tooling should use the `cors` trait on a service to automatically configure CORS on "
+                    + "clients and servers."),
```

Comment:
I think this should explicitly state that the apigateway integration will add this in. I know it's crossing the streams a bit, but I think it'll be helpful.

Reply:
Good idea. Updated.